### ISSUE TITLE
About vote option in test and noted data augmentation

### ISSUE THREAD

**Issue by a18700** [yanx27/Pointnet_Pointnet2_pytorch#46]
Hello.

I have questions as mentioned in title.

1. how the options of 'num_vote' are used for reporting your results?

2. and is the implementation of 'num_vote' are appropriate? does the randomness are inside the 'classifier' instance?

cf. original implementation of pointnet++ of authers include randomness when loading test data.

3. what is your additional data sugmentation trick mentioned in README.md?

If answered, it will be much more plausible to others.

thank you.

**Comment by yanx27** [comment]
@a18700 Hi, I actually use default setting in code for voting test.  Voting exactly will introduce randomness to experiment results, but it just a slight change. Furthermore, voting test will not increase the performance and it works for more stable results, Finally, I add randomly drop points in trianing processing, which already used in official PointNet++ :)

**Comment by a18700** [comment]
How kind you are!

One thing what i want to clarify remaining, how does the performance of pointnet++(ssg, msg) is better than official version?

For ssg, 92.4 and 90.9(in paper) is huge difference.

So at first i thought it's heavily depending on augmentations, but you replied as you continue to use same augmentations in paper.

Can you explain more about this enhancement in performance?

Thanks.

**Comment by yanx27** [comment]
@a18700 This work uses almost all the settings of the official version except:
(1) The ball query function maybe slightly different from official one
(2) I use both group normalized xyz and raw xyz to aggregate local group's features.
So I really do not sure that which one will enhance the performance :)

**Comment by ZJU-PLP** [comment]
@yanx27 Hi, could you mind sharing the memory usage when you train pointnet2_sem_seg? I find that the memeory need 20GB while offiical tensorflow version only 6GB!
![image](https://user-images.githubusercontent.com/32757213/72306475-20c3c180-36b2-11ea-85df-657aed467267.png)
 Ps:the batch_size=16

**Comment by yanx27** [comment]
@ZJU-PLP Do you use same CUDA version and torch version with me?

**Comment by chenzhaiyu** [comment]
@ZJU-PLP @yanx27 I also encountered this issue (*memory leaks*) when setting `num_workers > 0`. The worker process will be killed after some iterations, especially because I don't have a large memory. This seems to be an issue introduced by the reference counting of Python. To mitigate it you can set `num_workers = 0`, which slows down your data IO for sure. See https://github.com/pytorch/pytorch/issues/13246.
